keel-sql/src/test/kotlin/com/netflix/spinnaker/keel/sql/SqlResourceRepositoryPeriodicallyCheckedTests.kt

```diff
@@ -122,6 +115,29 @@ internal object SqlResourceRepositoryPeriodicallyCheckedTests :
       }
     }
   }
+
+  fun versionedResourceTests() = rootContext<Fixture<Resource<ResourceSpec>, SqlResourceRepository>> {
+    fixture {
+      Fixture(factory, createAndStore, updateOne)
+    }
+
+    context("multiple versions of a resource exist") {
+      before {
+        createAndStore(1)
+        updateOne()
+      }
+
+      test("the older resource version is never checked") {
+        expectThat(nextResults())
+          .hasSize(1)
+          .first()
+          .get(Resource<*>::version) isEqualTo 2
+
+        expectThat(nextResults())
+          .isEmpty()
+      }
+    }
+  }
```

Comment:
Our resource check loop should only check the latest versions of resources, not older ones